presto-main/src/test/java/io/prestosql/sql/planner/TestDomainCoercer.java

```diff
@@ -0,0 +1,166 @@
+    }
+
+    @Test
+    public void testNone()
+    {
+        assertEquals(applySaturatedCasts(Domain.none(BIGINT), INTEGER), Domain.none(INTEGER));
+    }
+
+    @Test
+    public void testAll()
+    {
+        assertEquals(applySaturatedCasts(Domain.all(BIGINT), INTEGER), Domain.all(INTEGER));
+    }
+
+    @Test
+    public void testOnlyNull()
+    {
+        assertEquals(applySaturatedCasts(Domain.onlyNull(BIGINT), INTEGER), Domain.onlyNull(INTEGER));
+    }
+
+    @Test
+    public void testCoercedValueSameAsOriginal()
+    {
+        assertEquals(
+                applySaturatedCasts(multipleValues(BIGINT, ImmutableList.of(1L, 10000L, -2000L)), SMALLINT),
+                multipleValues(SMALLINT, ImmutableList.of(1L, 10000L, -2000L)));
+
+        Domain original = Domain.create(
+                ValueSet.ofRanges(
+                        lessThan(DOUBLE, 0.0),
+                        range(DOUBLE, 0.0, false, 1.0, false),
+                        range(DOUBLE, 2.0, true, 3.0, true),
+                        greaterThan(DOUBLE, 4.0)),
+                true);
+        assertEquals(
+                applySaturatedCasts(original, REAL),
+                Domain.create(
+                        ValueSet.ofRanges(
+                                lessThan(REAL, (long) Float.floatToIntBits(0.0f)),
+                                range(REAL, (long) Float.floatToIntBits(0.0f), false, (long) Float.floatToIntBits(1.0f), false),
+                                range(REAL, (long) Float.floatToIntBits(2.0f), true, (long) Float.floatToIntBits(3.0f), true),
+                                greaterThan(REAL, (long) Float.floatToIntBits(4.0f))),
+                        true));
+    }
+
+    @Test
+    public void testOutsideTargetTypeRange()
+    {
+        assertEquals(
+                applySaturatedCasts(
+                        multipleValues(BIGINT, ImmutableList.of(1L, 10000000000L, -2000L)), SMALLINT),
+                multipleValues(SMALLINT, ImmutableList.of(1L, -2000L)));
+
+        assertEquals(
+                applySaturatedCasts(
+                        Domain.create(
+                                ValueSet.ofRanges(range(DOUBLE, 0.0, true, ((double) Float.MAX_VALUE) * 10, true)),
+                                true),
+                        REAL),
+                Domain.create(
+                        ValueSet.ofRanges((greaterThanOrEqual(REAL, (long) Float.floatToIntBits(0.0f)))),
+                        true));
+
+        assertEquals(
+                applySaturatedCasts(
+                        Domain.create(
+                                ValueSet.ofRanges(
+                                        range(DOUBLE, ((double) Float.MAX_VALUE) * -2, true, ((double) Float.MAX_VALUE) * 10, true),
+                                        range(DOUBLE, 0.0, false, 1.0, false)),
+                                true),
+                        REAL),
+                Domain.all(REAL));
+
+        assertEquals(
```

Comment:
add a test case where both high and low are below and over target range

Reply:
I tried to test this case in last assertion through 
`range(BIGINT, ((long) Integer.MAX_VALUE) * -2, false, ((long) Integer.MAX_VALUE) * 10, false)`
Did you have something else in mind ?

Reply:
I mean both low and hi should be either very negative (out of range) or very positive (out of range).

Reply:
added both those type of tests now